ophyd/status.py

```diff
@@ -93,6 +93,9 @@ def _settle_then_run_callbacks(self, success=True):
                 self._cb = None
 
     def _finished(self, success=True, **kwargs):
+        if self.done:
+            return
```

Comment:
I think this may produce a code path where the callback does not get run.  As `MoveStatus` can take a `done` kwarg (which tracing this looks like it gets reset back to false in the `super.__init__`, which makes be think there is another bug here.